examples/triangle_intersection/triangle_intersection.cpp

```diff
@@ -0,0 +1,387 @@
+  // actuallt inside the wrapped triangle and we have to check that here. This
+  // also gives us the opportunity to store the barycentric coordinates in case
+  // there is an intersection. Since the triangles don't overlap in this
+  // example, there is at most one triangle that contains a given point and we
+  // can abort the search early when we found a match.
+  template <typename Query, typename Primitive>
+  KOKKOS_FUNCTION auto operator()(Query const &query,
+                                  Primitive const &primitive) const
+  {
+    ArborX::ExperimentalHyperGeometry::Point<2> const &point =
+        getGeometry(getPredicate(query));
+    auto query_index = ArborX::getData(query);
+
+    auto const coeffs =
+        _triangles.get_mapping(primitive.index).get_coeff(point);
+    bool intersects = coeffs[0] >= 0 && coeffs[1] >= 0 && coeffs[2] >= 0;
+
+    if (intersects)
+    {
+      _offsets(query_index) = primitive.index;
+      _coefficients(query_index) = coeffs;
+      return ArborX::CallbackTreeTraversalControl::early_exit;
+    }
+    return ArborX::CallbackTreeTraversalControl::normal_continuation;
+  }
+
+private:
+  Triangles<MemorySpace> _triangles;
+  Kokkos::View<int *, MemorySpace> _offsets;
+  Kokkos::View<ArborX::Point *, MemorySpace> _coefficients;
+};
+
+// Now that we have encapsulated the objects and queries to be used within the
+// Triangles class, we can continue with performing the actual search.
+int main()
+{
+  Kokkos::ScopeGuard guard;
+
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+  using MemorySpace = typename ExecutionSpace::memory_space;
+  ExecutionSpace execution_space;
+
+  std::cout << "Create grid with triangles.\n";
+  Triangles<MemorySpace> triangles(execution_space);
+  std::cout << "Triangles set up.\n";
+
+  std::cout << "Creating BVH tree.\n";
+  ArborX::BasicBoundingVolumeHierarchy<
+      MemorySpace, ArborX::Details::PairIndexVolume<
+                       ArborX::ExperimentalHyperGeometry::Box<2>>> const
+      tree(execution_space, triangles);
+  std::cout << "BVH tree set up.\n";
+
+  std::cout << "Create the points used for queries.\n";
+  Points<MemorySpace> points(execution_space);
+  std::cout << "Points for queries set up.\n";
+
+  std::cout << "Starting the queries.\n";
+  int const n = points.size();
+  Kokkos::View<int *, MemorySpace> offsets("Example::offsets", n);
+  Kokkos::View<ArborX::Point *, MemorySpace> coefficients(
+      "Example::coefficients", n);
+
+  tree.query(execution_space, points,
+             TriangleIntersectionCallback<MemorySpace>{triangles, offsets,
+                                                       coefficients});
+  std::cout << "Queries done.\n";
+
+#ifndef NDEBUG
+  std::cout << "Starting checking results.\n";
+  Kokkos::parallel_for(
+      Kokkos::RangePolicy<ExecutionSpace>(execution_space, 0, n),
+      KOKKOS_LAMBDA(int i) {
+        constexpr float eps = 1.e-3;
```

Comment:
How was this eps chosen? What happens with smaller values?

Reply:
IIRC, this was about the magnitude I needed to pass CI.